sources/packages/web/src/views/aest/institution/ViewOffering.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <full-page-container>
+    <template #header>
+      <header-navigator
+        title="Study period offerings"
+        :routeLocation="programRoute"
+        subTitle="View Request"
+      >
+        <template #buttons v-if="showActionButtons">
+          <v-btn
+            color="primary"
+            variant="outlined"
+            @click="assessOffering(OfferingStatus.Declined)"
+            >Decline</v-btn
+          >
+          <v-btn
+            class="ml-2"
+            color="primary"
+            @click="assessOffering(OfferingStatus.Approved)"
+            >Approve offering</v-btn
+          >
+        </template>
+      </header-navigator>
+      <program-offering-detail-header
+        class="m-4"
+        :headerDetails="{
+          ...initialData,
+          status: initialData.offeringStatus,
+          institutionId: institutionId,
+        }"
+      />
+    </template>
+    <offering-form :data="initialData"></offering-form>
+    <assess-offering-modal
+      ref="assessOfferingModalRef"
+      :offeringStatus="offeringApprovalStatus"
+    />
+  </full-page-container>
+</template>
+
+<script lang="ts">
+import { EducationProgramOfferingService } from "@/services/EducationProgramOfferingService";
+import { EducationProgramService } from "@/services/EducationProgramService";
+import { onMounted, ref, computed } from "vue";
+import { OfferingFormBaseModel, OfferingStatus } from "@/types";
+import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
+import { useSnackBar, ModalDialog } from "@/composables";
+import { OfferingAssessmentAPIInDTO } from "@/services/http/dto";
+import { BannerTypes } from "@/components/generic/Banner.models";
+import ProgramOfferingDetailHeader from "@/components/common/ProgramOfferingDetailHeader.vue";
+import OfferingForm from "@/components/common/OfferingForm.vue";
+import AssessOfferingModal from "@/components/aest/institution/modals/AssessOfferingModal.vue";
+
+export default {
+  components: {
+    ProgramOfferingDetailHeader,
+    OfferingForm,
+    AssessOfferingModal,
+  },
+  props: {
+    institutionId: {
+      type: Number,
+      required: true,
+    },
+    locationId: {
+      type: Number,
+      required: true,
+    },
+    programId: {
```

Comment:
As mentioned before, receiving the `programId` in the URL opens for URL manipulation.

Reply:
Makes sense, If possible will try to do in the next PR otherwise will create a tech debt for it.